src/services/Store/Contact/helpers.ts

```diff
@@ -68,3 +77,32 @@ export const findMultipleNextUnusedDefaultLabels = (wallet: WalletId, numOfLabel
 
 export const findNextRecipientLabel = (contacts: Contact[]) =>
   getUnusedLabel(contacts, (index) => `Recipient ${index}`);
+
+export const getContactFromContracts = (contracts: ExtendedContract[]) => (
+  address: TAddress,
+  networkId?: NetworkId
+): ExtendedContact | undefined => {
+  const contract = contracts.find(
+    (c) => c.address === address && (c.networkId === networkId || !networkId)
+  );
+  const contact: ExtendedContact | undefined = contract && {
+    address,
+    label: contract.name,
+    network: contract.networkId,
+    notes: '',
+    uuid: contract.uuid
+  };
+  return contact;
+};
+
+export const getContactByAddressAndNetworkId = (
+  contacts: ExtendedContact[],
+  contracts: ExtendedContract[]
+) => (address: TAddress, networkId: NetworkId) => {
+  return (
+    [...contacts, ...STATIC_CONTACTS]
+      .filter((contact: ExtendedContact) => contact.network === networkId)
+      .find((contact: ExtendedContact) => isSameAddress(contact.address as TAddress, address)) ||
+    getContactFromContracts(contracts)(address, networkId)
+  );
+};
```

Comment:
Can't this be done in redux ?

Reply:
Yes, but as discussed in DMs that would be out of scope for this PR.